### ISSUE TITLE
[Feature Request]Synchronized batch norm

### ISSUE THREAD

**Issue by labor00** [pytorch/pytorch#12198]
## 🚀 Feature
Synchronized batch norm. Batch norm operation that synchronizes batch statistics across devices.

## Motivation
Many tasks specially ones using FCN's for  instance/semantic segmentation requires a significant amount of GPU memory. Distributing the batch elements across devices is a common way to train these models. However due to the lack of synchronization of batch statistics this procedure can run into problems.
There is at least one issue previously opened (#2584) about this feature.

## Alternatives
There are also some (and maybe others) community options for instance:

[inplace_abn](https://github.com/mapillary/inplace_abn):  This repository offers synchronized  inplace ABN to reduce memory footprint (combines both batch norm and activation in a single node). However the operator fusion procedure restrict the type of activation used, for instance is not possible to use ReLU activation. (Currently broken with pytorch master).

[encoding](https://hangzhang.org/PyTorch-Encoding/nn.html#batchnorm2d): Python only synchronized batch norm operation.

There are also other options for instance using [group norm](https://arxiv.org/abs/1803.08494) but this is not always possible or desirable. For instance most of the pretrained models used as encoder on these problems only uses batch norm.

**Comment by lysukhin** [comment]
Any updates on this?

**Comment by soumith** [comment]
Added now in PyTorch master via https://github.com/pytorch/pytorch/pull/14267
For documentation, see: https://pytorch.org/docs/master/nn.html#torch.nn.SyncBatchNorm

**Comment by cardoso-neto** [comment]
If anyone happens to be struggling to get the `DistributedDataParallel` to use `SyncBatchnorm`, take a look at this small step-by-step a colleague of mine wrote: [github/dougsouza/pytorch-sync-batchnorm-example](https://github.com/dougsouza/pytorch-sync-batchnorm-example). Hope it helps. :)
